web/src/__tests__/alert-node.spec.ts

```diff
@@ -0,0 +1,66 @@
+    const expectedKorrel8rQuery =
+      'alert:alert:{"alertname":"KubePodCrashLooping","container":"bad-deployment","namespace":"default","pod":"bad-deployment-000000000-00000"}';
+    expect(AlertNode.fromURL(url)?.toQuery()).toEqual(expectedKorrel8rQuery);
+  });
+
+  it('Test query to url parsing', () => {
+    const query =
+      'alert:alert:{"alertname":"KubePodCrashLooping","container":"bad-deployment","namespace":"default","pod":"bad-deployment-000000000-00000"}';
+    const expectedKorrel8rURL =
+      'monitoring/alerts?alertname=KubePodCrashLooping&container=bad-deployment&namespace=default&pod=bad-deployment-000000000-00000&name=KubePodCrashLooping';
+    expect(AlertNode.fromQuery(query)?.toURL()).toEqual(expectedKorrel8rURL);
+  });
+
+  it('Query => URL => Query', () => {
+    const query =
+      'alert:alert:{"alertname":"KubePodCrashLooping","container":"bad-deployment","namespace":"default","pod":"bad-deployment-000000000-00000"}';
+    const expectedKorrel8rURL =
+      'monitoring/alerts?alertname=KubePodCrashLooping&container=bad-deployment&namespace=default&pod=bad-deployment-000000000-00000&name=KubePodCrashLooping';
+    const actualKorrel8rURL = AlertNode.fromQuery(query)?.toURL();
+    expect(actualKorrel8rURL).toEqual(expectedKorrel8rURL);
+    expect(AlertNode.fromURL(actualKorrel8rURL)?.toQuery()).toEqual(query);
+  });
+
+  it('URL => Query => URL', () => {
+    const url =
+      'monitoring/alerts?alertname=KubePodCrashLooping&container=bad-deployment&namespace=default&pod=bad-deployment-000000000-00000&name=KubePodCrashLooping';
+    const expectedQuery =
+      'alert:alert:{"alertname":"KubePodCrashLooping","container":"bad-deployment","namespace":"default","pod":"bad-deployment-000000000-00000"}';
+    const actualQuery = AlertNode.fromURL(url)?.toQuery();
+    expect(actualQuery).toEqual(expectedQuery);
+    expect(AlertNode.fromQuery(expectedQuery)?.toURL()).toEqual(url);
+  });
+
+  it('Test url to query parsing with expected errors', () => {
+    let url = 'monitoring/alert';
+    expect(() => AlertNode.fromURL(url)).toThrow('Expected url to start with monitoring/alerts');
+    url = url.concat('s');
+    expect(() => AlertNode.fromURL(url)).toThrow('Expected URL to contain query parameters');
+    url = url.concat('?');
+    expect(() => AlertNode.fromURL(url)).toThrow('Expected URL to contain query parameters');
+    url = url.concat('prometheus=openshift-monitoring/k8s');
+    expect(() => AlertNode.fromURL(url)).toThrow('Expected more than 0 relevant query parameters');
+  });
+
+  it('Test query to url parsing with expected errors', () => {
+    let query = 'alert:aler';
+    expect(() => AlertNode.fromQuery(query)).toThrow('Expected query to start with alert:alert:');
+    query = query.concat('t:');
```

Comment:
same case as above

Reply:
Updated test cases to reflect this new pattern, and removed some of the test cases which were redundant with the X => Y => X testing